components/buildless-serverless/internal/controller/cache/cache.go

```diff
@@ -0,0 +1,63 @@
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+var (
+	_ InMemoryCache = (*inMemoryCache)(nil)
+)
+
+type InMemoryCache interface {
+	Set(any, string)
+	Get(any) string
+	Delete(any)
+}
+
+// inMemoryCache provides an in-memory processor to store buildless serverless key values pairs with timestamp. By using sync.Map for caching,
+// concurrent operations to the processor from diverse reconciliations are considered safe.
+//
+// Inside the processor is stored last commit with which was git function created by repository url and reference as the key.
+type inMemoryCache struct {
```

Comment:
I suggest thinking of a better, more precise name for this struct. Maybe `repoLastCommitCache`? This cache is not generic enough to call it just `inMemoryCache`.

Reply:
Thanks for that, changed as suggested to `repoLastCommitCache`